### ISSUE TITLE
Error response from daemon: Cannot start container: [8] System error: no such file or directory

### ISSUE THREAD

**Issue by bezelga** [docker/docker#14931]
I just installed experimental docker and I get this error while trying to run hello world:

```docker run hello-world```

I get this:

```
Error response from daemon: Cannot start container 53aa09a280d8d60d013109355669573c2fdbcc4c0ee15447c2b563ee2e18e947: [8] System error: no such file or directory
```

version:

```
Client:
 Version:      1.8.0-dev
 API version:  1.20
 Go version:   go1.4.2
 Git commit:   e5d8fb9
 Built:        Thu Jul 23 17:51:33 UTC 2015
 OS/Arch:      linux/amd64
 Experimental: true

Server:
 Version:      1.8.0-dev
 API version:  1.20
 Go version:   go1.4.2
 Git commit:   e5d8fb9
 Built:        Thu Jul 23 17:51:33 UTC 2015
 OS/Arch:      linux/amd64
 Experimental: true
```

**Comment by ewindisch** [comment]
Any chance you installed from the debian packages?

**Comment by bezelga** [comment]
Not sure, I am in a Ubuntu 14.04 and did:

``` wget -qO- https://experimental.docker.com/ | sh```

**Comment by ewindisch** [comment]
@bezelga you're likely running into a bug with the apparmor profiles fixed in #14934. You can confirm by seeing if you have DENIED entries in your audit log (dmesg or, if you have auditd installed, in /var/log/audit/audit.log)

**Comment by ewindisch** [comment]
@bezelga **If (and only if)** you see those DENIED messages (especially if for auplink), then try checking out my PR and loading my apparmor policy:

```
git fetch origin pull/14934/head:apparmor-engine-policy-nofile
git checkout apparmor-engine-policy-nofile
apparmor_parser -r contrib/apparmor/docker-engine
```

If this works you can copy the contrib/apparmor/docker-engine file into /etc/apparmor.d. If not, then you could try loading it in "complain" mode by running 'apparmor_parser -r -C'. That will generate logs for debugging.

**Comment by stewartpark** [comment]
I'm having the same result here.

```
root@swarm-sf-0:~# docker version
Client:
 Version:      1.8.0-dev
 API version:  1.20
 Go version:   go1.4.2
 Git commit:   e5d8fb9
 Built:        Thu Jul 23 17:51:33 UTC 2015
 OS/Arch:      linux/amd64
 Experimental: true

Server:
 Version:      1.8.0-dev
 API version:  1.20
 Go version:   go1.4.2
 Git commit:   e5d8fb9
 Built:        Thu Jul 23 17:51:33 UTC 2015
 OS/Arch:      linux/amd64
 Experimental: true
root@swarm-sf-0:~# docker info
Containers: 0
Images: 7
Storage Driver: aufs
 Root Dir: /var/lib/docker/aufs
 Backing Filesystem: extfs
 Dirs: 7
 Dirperm1 Supported: true
Execution Driver: native-0.2
Logging Driver: json-file
Kernel Version: 3.16.0-41-generic
Operating System: Ubuntu 14.04.2 LTS
CPUs: 1
Total Memory: 489.6 MiB
Name: swarm-sf-0
ID: SNDU:RD22:KOVH:IEBT:2VTC:7MKS:5RIP:QG7S:NIW5:W6YA:HJBD:SXRM
WARNING: No swap limit support
Labels:
 com.docker.network.driver.overlay.bind_interface=eth0
Experimental: true
root@swarm-sf-0:~# uname -a
Linux swarm-sf-0 3.16.0-41-generic #57-Ubuntu SMP Thu Jun 18 08:44:16 UTC 2015 x86_64 x86_64 x86_64 GNU/Linux
```

Seems like it got broken very recently, because it's been working fine when I tried to install the experimental version just a day ago.

The below is the working experimental version(Git commit: 8c7cd78) i installed a few days ago. (It still works fine)

```
Client:
 Version:      1.8.0-dev
 API version:  1.20
 Go version:   go1.4.2
 Git commit:   8c7cd78
 Built:        Tue Jul 14 23:47:18 UTC 2015
 OS/Arch:      linux/amd64
 Experimental: true

Server:
 Version:      1.8.0-dev
 API version:  1.20
 Go version:   go1.4.2
 Git commit:   8c7cd78
 Built:        Tue Jul 14 23:47:18 UTC 2015
 OS/Arch:      linux/amd64
 Experimental: true
```

I hope this helps the investigation on what broke it.

And also, is there any way to install a previous version of the experimental docker? If so, please let me know. :)

**Comment by bezelga** [comment]
@ewindisch 

looking into `/var/log/dmesg` I did not find any `DENIED` message.


these are the lines that contain apparmor stuff:

```
[    2.344198] [drm] Initialized drm 1.1.0 20060810
[    2.415556] type=1400 audit(1437680709.158:2): apparmor="STATUS" operation="profile_load" profile="unconfined" name="/sbin/dhclient" pid=400 comm="apparmor_parser"
[    2.415562] type=1400 audit(1437680709.158:3): apparmor="STATUS" operation="profile_load" profile="unconfined" name="/usr/lib/NetworkManager/nm-dhcp-client.action" pid=400 comm="apparmor_parser"
[    2.415566] type=1400 audit(1437680709.158:4): apparmor="STATUS" operation="profile_load" profile="unconfined" name="/usr/lib/connman/scripts/dhclient-script" pid=400 comm="apparmor_parser"
[    2.415966] type=1400 audit(1437680709.158:5): apparmor="STATUS" operation="profile_replace" profile="unconfined" name="/usr/lib/NetworkManager/nm-dhcp-client.action" pid=400 comm="apparmor_parser"
[    2.415971] type=1400 audit(1437680709.158:6): apparmor="STATUS" operation="profile_replace" profile="unconfined" name="/usr/lib/connman/scripts/dhclient-script" pid=400 comm="apparmor_parser"
[    2.416213] type=1400 audit(1437680709.162:7): apparmor="STATUS" operation="profile_replace" profile="unconfined" name="/usr/lib/connman/scripts/dhclient-script" pid=400 comm="apparmor_parser"
[    2.632079] [TTM] Zone  kernel: Available graphics memory: 250896 kiB
[    2.632079] [TTM] Initializing pool allocator
[    2.632083] [TTM] Initializing DMA pool allocator
[    2.632327] [drm] fb mappable at 0x0
[    2.632328] [drm] vram aper at 0x0
[    2.632328] [drm] size 0
[    2.632329] [drm] fb depth is 24
[    2.632329] [drm]    pitch is 3072
[    2.712329] fbcon: cirrusdrmfb (fb0) is primary device
[    2.764490] Console: switching to colour frame buffer device 128x48
[    2.787916] cirrus 0000:00:02.0: fb0: cirrusdrmfb frame buffer device
[    2.787918] cirrus 0000:00:02.0: registered panic notifier
[    2.960455] [drm] Initialized cirrus 1.0.0 20110418 for 0000:00:02.0 on minor 0
[    5.012091] init: failsafe main process (702) killed by TERM signal
[    5.267239] type=1400 audit(1437680712.010:8): apparmor="STATUS" operation="profile_replace" profile="unconfined" name="/sbin/dhclient" pid=835 comm="apparmor_parser"
[    5.267248] type=1400 audit(1437680712.010:9): apparmor="STATUS" operation="profile_replace" profile="unconfined" name="/usr/lib/NetworkManager/nm-dhcp-client.action" pid=835 comm="apparmor_parser"
[    5.267252] type=1400 audit(1437680712.010:10): apparmor="STATUS" operation="profile_replace" profile="unconfined" name="/usr/lib/connman/scripts/dhclient-script" pid=835 comm="apparmor_parser"
```

Let me know if there is anything else I can help.

In the meantime would be good to install that previous version @stewartpark mentioned.

**Comment by bezelga** [comment]
@stewartpark hello there, did you figure out how to install that working version? thanks

**Comment by ankushagarwal** [comment]
+1 
I am having the same problem. Installed using `wget -qO- https://experimental.docker.com/ | sh` on a fresh ubuntu 14.04 VM and I get the error. I am running the same version (`Git commit:   e5d8fb9`) as @stewartpark 

@ewindisch  I don't have the `/var/log/audit/audit.log` file on my system and I don't see any `denied` message in `/var/log/dmesg`

**Comment by JeanMertz** [comment]
+1, exact same situation/outcome as @ankushagarwal

**Comment by bezelga** [comment]
It worked for me in this experimental version:

```docker version
Client:
 Version:      1.8.0-dev
 API version:  1.20
 Go version:   go1.4.2
 Git commit:   a2ea8f2
 Built:        Sat Jul 25 22:03:05 UTC 2015
 OS/Arch:      linux/amd64
 Experimental: true

Server:
 Version:      1.8.0-dev
 API version:  1.20
 Go version:   go1.4.2
 Git commit:   a2ea8f2
 Built:        Sat Jul 25 22:03:05 UTC 2015
 OS/Arch:      linux/amd64
 Experimental: true```

**Comment by samek** [comment]
For me it's happening on 1.7.1 

     docker run -ti ubuntu /bin/bash
     no such file or directory
     Error response from daemon: Cannot start container      0e43eff54a6ad1ced982a00049bc5ba8679a35b02d1d216286c8a1d96579f8fa: [8] System error: no such file or directory
     
     docker version
     Client version: 1.7.1
     Client API version: 1.19
     Go version (client): go1.4.2
     Git commit (client): 786b29d
     OS/Arch (client): linux/amd64
     Server version: 1.7.1
     Server API version: 1.19
     Go version (server): go1.4.2
     Git commit (server): 786b29d     
     OS/Arch (server): linux/ad64


     cat /etc/issue.net 
     Ubuntu 12.04.5 LTS
     uname -a
     Linux enc2 3.13.0-53-generic #89~precise1-Ubuntu SMP Wed May 20 17:42:16 UTC 2015 x86_64 x86_64 x86_64 GNU/Linux

**Comment by djmaze** [comment]
I've got an image which is hosted on the registry hub. On 4 out of 5 CoreOS machines, it can be pulled and run correctly. On the 5th machine, there is the same error as reported here when running the image. The machines are set up identically (CoreOS stable 717.3.0).

As this is an automated build, I tried rebuilding it. The new image has the same problem, but again, only on this one specific machine.

In the image folder in `/var/lib/docker/overlay/5c01709e62f2a1d95560ff7b046e86b54ed1a57011bacbd52043d2dcd6c75ca0/root/` on the misbehaving server, I can see the following contents: 

```
-rwxr-xr-x 3 root root  612 Aug  5 11:38 custom-entrypoint.sh
-rwxr-xr-x 7 root root 3110 Jun 17 22:15 entrypoint.sh
drwxr-xr-x 6 root root 4096 Aug  5 11:46 etc
drwxr-xr-x 3 root root 4096 Aug  5 11:46 lib
drwx------ 3 root root 4096 Aug  5 11:46 root
drwx------ 2 root root 4096 Jul 27 14:21 temp-docker-extract009171261
drwxr-xr-x 6 root root 4096 Aug  5 11:46 usr
drwxr-xr-x 5 root root 4096 Aug  5 11:46 var
```

The same image on the other servers has a complete filesystem. The existence of the (empty) folder `temp-docker-extract009171261` tells me there must have been some error during the pull. Which is confusing since there was no error message in the logs. Even with debug logging enabled, there is nothing particularly striking to see. Apart from the (debug-level) message `Verification failed for xxx using key yyy..` 

Got this working using a workaround now. As we are using a private registry, too (using the `--insecure-registry` switch btw), I tried pushing the intact image from one of the other servers up there. Pulling that image from the misbehaving server yields a sane, runnable image again. Not sure what to make out this.

```
Client version: 1.6.2
Client API version: 1.18
Go version (client): go1.4.2
Git commit (client): 7c8fca2-dirty
OS/Arch (client): linux/amd64
Server version: 1.6.2
Server API version: 1.18
Go version (server): go1.4.2
Git commit (server): 7c8fca2-dirty
OS/Arch (server): linux/amd64

Storage Driver: overlay
 Backing Filesystem: extfs
Execution Driver: native-0.2
Kernel Version: 4.0.5
```
